src/main/cpp/src/bloom_filter.cu

```diff
@@ -0,0 +1,148 @@
+                                     cudf::size_type num_hashes)
+{
+  int const tid = threadIdx.x + blockIdx.x * blockDim.x;
+  if (tid >= input.size()) { return; }
+
+  // https://github.com/apache/spark/blob/7bfbeb62cb1dc58d81243d22888faa688bad8064/common/sketch/src/main/java/org/apache/spark/util/sketch/BloomFilterImpl.java#L87
+  bloom_hash_type const h1 = MurmurHash3_32<int64_t>(0)(input[tid]);
+  bloom_hash_type const h2 = MurmurHash3_32<int64_t>(h1)(input[tid]);
+
+  // set a bit in the bloom filter for each hashed value
+  for (auto idx = 1; idx <= num_hashes; idx++) {
+    bloom_hash_type combined_hash = h1 + (idx * h2);
+
+    auto const [word_index, mask] = gpu_get_hash_mask(combined_hash, bloom_filter_bits);
+    atomicOr(bloom_filter + word_index, mask);
+  }
+}
+
+struct bloom_probe_functor {
+  cudf::bitmask_type const* const bloom_filter;
+  cudf::size_type const bloom_filter_bits;
+  cudf::size_type const num_hashes;
+
+  __device__ bool operator()(int64_t input)
+  {
+    // https://github.com/apache/spark/blob/7bfbeb62cb1dc58d81243d22888faa688bad8064/common/sketch/src/main/java/org/apache/spark/util/sketch/BloomFilterImpl.java#L110
+    // this code could be combined with the very similar code in gpu_bloom_filter_put. i've
+    // left it this way since the expectation is that we will early out fairly often, whereas
+    // in the build case we never early out so doing the additional if() return check is pointless.
+    bloom_hash_type const h1 = MurmurHash3_32<int64_t>(0)(input);
+    bloom_hash_type const h2 = MurmurHash3_32<int64_t>(h1)(input);
+
+    // set a bit in the bloom filter for each hashed value
+    for (auto idx = 1; idx <= num_hashes; idx++) {
+      bloom_hash_type combined_hash = h1 + (idx * h2);
+      auto const [word_index, mask] = gpu_get_hash_mask(combined_hash, bloom_filter_bits);
+      if (!(bloom_filter[word_index] & mask)) { return false; }
+    }
+    return true;
+  }
+};
+
+}  // anonymous namespace
+
+std::unique_ptr<rmm::device_buffer> bloom_filter_create(int64_t bloom_filter_bits,
+                                                        rmm::cuda_stream_view stream,
+                                                        rmm::mr::device_memory_resource* mr)
+{
+  std::unique_ptr<rmm::device_buffer> out = std::make_unique<rmm::device_buffer>(
+    cudf::num_bitmask_words(bloom_filter_bits) * sizeof(cudf::bitmask_type), stream, mr);
+  cudaMemsetAsync(out->data(), 0, out->size(), stream);
+  return out;
+}
+
+void bloom_filter_put(cudf::device_span<cudf::bitmask_type> bloom_filter,
+                      int64_t bloom_filter_bits,
+                      cudf::column_view const& input,
+                      cudf::size_type num_hashes,
+                      rmm::cuda_stream_view stream)
+{
+  CUDF_EXPECTS(input.type() == cudf::data_type{cudf::type_id::INT64} && !input.nullable(),
+               "bloom filter input expects a non-nullable column of int64s");
```

Comment:
Can we replace the interface with `device_uvector<int64_t>`?

Reply:
Which interface? `gpu_bloom_filter_put`? The reason I don't like a span there is that it's passing useless information.  We don't need to know the size of the `bloom_filter` array itself.  We just care how many bits there are (which, importantly, can be different from length * 32)